tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -1428,16 +1441,21 @@ void CorProfiler::AddDerivedInstrumentations(WCHAR* id, CallTargetDefinition* it
     }
 }
 
-void CorProfiler::InternalAddInstrumentation(WCHAR* id, CallTargetDefinition* items, int size, bool isDerived)
+void CorProfiler::InternalAddInstrumentation(WCHAR* id, CallTargetDefinition* items, int size, bool isDerived, bool enable)
 {
     shared::WSTRING definitionsId = shared::WSTRING(id);
     std::scoped_lock<std::mutex> definitionsLock(definitions_ids_lock_);
 
-    if (definitions_ids_.find(definitionsId) != definitions_ids_.end())
+    if (enable && definitions_ids_.find(definitionsId) != definitions_ids_.end())
     {
         Logger::Info("InitializeProfiler: Id already processed.");
         return;
     }
+    if (!enable && definitions_ids_.find(definitionsId) == definitions_ids_.end())
```

Comment:
this is silly but, we can do just one `const auto isDefined = definitions_ids_.find(definitionsId) != definitions_ids_.end()` and use it for both comparisons.

Reply:
True. I'll look into it

Reply:
Fixed in e3ca95392f32b1632ba61e0b78f15421b385ab85